smart_campus/app/views.py

```diff
@@ -1477,7 +1473,7 @@ def activate(request, uidb64, token):
 @csrf_exempt
 @require_POST
 @activate_required
-def reset_password(request):
+def reset_password(request, email):
```

Comment:
Consider using token instead of using email directly.
flasky does a good job on it.
I suggest you take a look on [register](https://github.com/miguelgrinberg/flasky/blob/master/app/auth/views.py#L51) and [confirm](https://github.com/miguelgrinberg/flasky/blob/master/app/auth/views.py#L69) views.

Reply:
Thanks a lot for the recommendation, But I don't get your point.
Can you explain more detail about the flask way?

Reply:
Encrypt the information as token, send it in the URL and then decrypt it in the api.

Reply:
Still confused...
I done for it with a `token.py` file, and use self-defined to valid a activation like: `account_activation_token.make_token(user)`

And use default django token function with the reset password part (`default_token_generator.make_token(request.user)`), and decrypt it in the reset_password_page (`default_token_generator.check_token(user, token)`)

Reply:
Did you mean that we should encrypt the api url with reset_password request?
If so, I think the encryption part is a front-end job (and decrytpion is a back-end job), And we should negotiate about the encryption way.

Reply:
Oh, I see your point.
It seems I misunderstood.
But why do you have to embed email in URL?
Can't you get it from `request.user` or post data?

Reply:
Because the form-data is a new method for web browser and android under version 5 can't use this method to pass the argument.
We (involved front-end) are still negotiating that how to refactor other api part, but for this consideration I choose pass the email in the url pattern.

Reply:
And I looks at some example at the internet like stackoverflow, it seems a reasonable way to pass the email with url?

Reply:
I see your point.
It sounds reasonable.